src/js/components/Select/__tests__/Select-test.js

```diff
@@ -1592,5 +1589,76 @@ describe('Select', () => {
     expect(asFragment()).toMatchSnapshot();
   });
 
+  test('opens drop with click but focus option with keyboard', () => {
+    jest.useFakeTimers();
+    const onSearch = jest.fn();
+    const { getAllByRole, getByPlaceholderText } = render(
+      <Grommet>
+        <Select
+          id="test-select"
+          options={['one', 'two']}
+          onSearch={onSearch}
+          placeholder="test select"
+        />
+      </Grommet>,
+    );
+    const select = getByPlaceholderText('test select');
+    fireEvent.click(select);
```

Comment:
@jcfilben Do you feel that this test is a candidate to use userEvent instead of fireEvent?

I know userEvent is preferred as a best practice but also has some shortcomings. Wanted to get your opinion before suggesting the PR author to make any changes.

Reply:
Good question, I think places where we use useFakeTimers we've run into issues with `userEvent`. But there is a chance that `userEvent` could work here. I haven't nailed down the exact situations that it does and doesn't work I just know useFakeTimers + userEvent has caused issues on other tests

Reply:
Thanks, Jessica. @miqh would you be open to trying `userEvent` in place of `fireEvent` in your tests? Let us know if you run into any issues related to the points Jessica mentioned above

Reply:
@taysea, I've given it a shot, but to no ease of success.

The best I could arrive at was something like the following:

```js
// Test itself is marked async
const user = userEvent.setup();
...
const select = getByPlaceholderText('test select');
await user.click(select);
await waitFor(() => expect(getAllByRole('searchbox')[0]).toHaveFocus());
// Select does not respond to navigation if event is directed at the search input
await user.type(document.getElementById('test-select__select-drop'), '[ArrowDown]');
// Search input does unfocus
await waitFor(() => expect(getAllByRole('searchbox')[0]).not.toHaveFocus());
// But the focus is instead directed at a container element and not the first option
await waitFor(() => expect(getAllByRole('option')[0]).toHaveFocus());
```
I should also add that I tried `user.keyboard`, which fell short of even being able to shift focus (i.e. respond to the arrow down) once the drop appears.

This is presumably because the latter acts on the element in focus. However, as I've noted, dispatching the arrow down key while the search input is in focus causes the Select to not respond (i.e. something must be swallowing up that event?).

If I contrast this with my existing test, that single `fireEvent.keyDown` invocation, plus a delay, was able to correctly (or at least for the purpose of the test) simulate a keyboard navigation step and to also focus the first option as expected.

Any suggestions on your part? I'm not sure what to do at this stage if there's insistence on `userEvent` being used instead.

Otherwise I'll update the bundle size test and call it a day.

Reply:
Hey @miqh thanks for trying `userEvent`, since it is not working I am okay with us sticking with the original test